src/libultra/io/aisetnextbuf.c

```diff
@@ -14,12 +17,15 @@ s32 osAiSetNextBuffer(void* buf, u32 size) {
         D_80130500 = false;
     }
 
+    // Originally a call to __osAiDeviceBusy
     status = HW_REG(AI_STATUS_REG, s32);
     if (status & AI_STATUS_AI_FULL) {
         return -1;
     }
 
-    HW_REG(AI_DRAM_ADDR_REG, u32) = PHYSICAL_TO_VIRTUAL(bufAdjusted);
+    // VIRTUAL_TO_PHYSICAL replaces osVirtualToPhysical, this replacement
+    // assumes that only KSEG0 addresses are given
+    HW_REG(AI_DRAM_ADDR_REG, u32) = VIRTUAL_TO_PHYSICAL(bufAdjusted);
```

Comment:
Since this is libultra, should it use `OS_K0_TO_PHYSICAL` to avoid depending on our own `VIRTUAL_TO_PHYSICAL` macro? I imagine it wouldn't make sense for a libultra file to include a project specific header either